### PR TITLE
Addresses #321 Fixes #464 Adds image search support for Yahoo

### LINKED ISSUE #464: Add Image search support of Yahoo
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behavior:**

Currently, No support of Image search in Yahoo.
**Expected behavior:**

Add Image search support of Yahoo
**Steps to reproduce:**

**Yes, I would like to work on this issue**

### PR DESCRIPTION
**Pull request by Remorax**
Fixes #464 
Addresses #321 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Image search has been added for Yahoo
- Code restructuring has been done so that image search for other websites can also be done in a similar manner

Link: https://enigmatic-dawn-30817.herokuapp.com/

Please review 😄

### CODE REVIEW

File: app/scrapers/yahoo.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from __future__ import print_function
 from .generalized import Scraper
-
+import re
```
Comment (cclauss) [inline app/scrapers/yahoo.py]: Why use __re.sub()__ when you can use [__'str'.replace()__](https://docs.python.org/3/library/stdtypes.html?#str.replace)?

**Comment by Remorax** [comment]
Again, I think its because of adding separate methods for Yahoo since the normal generalised procedure does not work here. I think this will be okay as more and more search engines are added, as the if condition will pass more often. Can this be merged, if there aren't any other issues? Thanks 😄

**Comment by cclauss** [comment]
@Remorax You want to add new specialized methods but not add tests for them?  I believe that @ParthS007 request that you add tests to cover the new functionality that you wish to add is both practical and resonable.

**Comment by bhaveshAn** [comment]
Please update the branch

**Comment by bhaveshAn** [comment]
Please fix the travis :+1:

**Comment by Remorax** [comment]
Not adding `@staticmethod` before `parse_image_response` fails Codacy, but adding it fails both Travis and Codacy because it gives missing positional argument as it doesn't get the self argument. What should I do?
Can I remove `@staticmethod` and can Codacy be ignored?

**Comment by bhaveshAn** [comment]
> Not adding @staticmethod before parse_image_response fails Codacy, but adding it fails both Travis and Codacy because it gives missing positional argument error as it isn't able to get the self argument. 

Please add @staticmethod and remove self from parameters of that function . I hope it will pass both the tests then.

**Comment by bhaveshAn** [comment]
Also update the branch because we are now following @staticmethod technique after #458

**Comment by cclauss** [comment]
@Remorax Really nice work on this PR!

**Pull request merged**